06-files-input-and-output/csv_split.py

```diff
@@ -0,0 +1,133 @@
+    if os.path.exists(file_path):
+        return True
+    else:
+        print("File not found:", file_path)
+        return False
+
+def validate_minimum_rows(csv_file_name, csv_file_num_lines, row_limit):
+    print(
+        "Validating file {file_name} has more than {row_limit} rows to split by...".format(
+            file_name=csv_file_name,
+            row_limit=row_limit,
+        ),
+        end=""
+    )
+
+    if csv_file_num_lines < row_limit:
+        print(
+            "Error: the file {file_name} contains {file_num_lines} lines, "
+            "it is smaller than the {row_limit} rows to split by".format(
+                file=csv_file_name,
+                file_num_lines=csv_file_num_lines,
+                row_limit= row_limit,
+            )
+        )
+        return False
+    else:
+        print("OK")
+        print("There are " + str(csv_file_num_lines) + " lines in the file")
+        return True
+
+def split_csv_to_chunks_by_rows(csv_data, row_limit):
+    print("Splitting CSV into chunks every " + str(row_limit) + " lines")
+
+    return split_list_in_groups_of(row_limit, csv_data)
+
+def split_list_in_groups_of(group_size, list_to_slice):
+    slices = list()
+
+    for index, item in enumerate(list_to_slice):
+        if index % group_size == 0:
+            current_slice = list()
+            current_slice.append(item)
+            slices.append(current_slice)
+        else:
+            current_slice.append(item)
+
+    return slices
+
+def read_file(file_path):
+    with open(file_path, "r") as file_handler:
+        return file_handler.readlines()
+
+def save_file(file_path, data):
+    with open(file_path, 'w') as file_handler:
+        file_handler.writelines(data)
+
+def output_to_files(csv_chunks, output_path):
+    header = csv_chunks[0].pop(0)
+    os.makedirs(output_path, exist_ok=True)
+
+    for chunk_number, chunk in enumerate(csv_chunks):
+        chunk.insert(0, header)
+        file_name = "csv_part_" + str(chunk_number)
+        output_file = os.path.join(output_path, file_name)
+        print("Saving {file_name} with {lines} lines in it".format(
+            file_name=output_file,
+            lines=len(chunk),
+        ))
+        save_file(output_file, chunk)
+
+def main():
+    options = get_arguments()
+    try:
+        if options:
+            if check_if_file_exists(options.input_file):
+                csv_data = read_file(options.input_file)
+                csv_lines = len(csv_data)
+                is_valid = validate_minimum_rows(
+                    csv_file_name=options.input_file,
+                    csv_file_num_lines=csv_lines,
+                    row_limit=options.row_limit,
+                )
+            else:
+                is_valid = False
+
+            if is_valid:
+                csv_chunks = split_csv_to_chunks_by_rows(csv_data, options.row_limit)
+                output_to_files(csv_chunks, options.output_path)
+    except Exception as e:
```

Comment:
Generally I don't like to have catch Exception since it's too general. The catch should handle expected cases, that is expected exceptions that might be raised. If an unexpected error is raised it's a bug and should be handled separately. Try to think where an error might be raised in the code and why when writing a try/except clause.